src/badguy/zeekling.cpp

```diff
@@ -17,28 +17,36 @@
 
 #include "badguy/zeekling.hpp"
 
-#include <math.h>
-
+#include "math/easing.hpp"
 #include "math/random.hpp"
+#include "math/vector.hpp"
+#include "math/util.hpp"
 #include "object/player.hpp"
 #include "sprite/sprite.hpp"
+#include "supertux/sector.hpp"
+
+const float FLYING_SPEED = 180.f;
+const float CHARGING_SPEED = 150.f;
+const float DIVING_SPEED = 280.f;
+
+const float CHARGING_DURATION = 0.3f;
+const float DIVING_DURATION = 1.5f;
+const float RECOVER_DURATION = 2.8f;
 
 Zeekling::Zeekling(const ReaderMapping& reader) :
   BadGuy(reader, "images/creatures/zeekling/zeekling.sprite"),
-  speed(gameRandom.randf(130.0f, 171.0f)),
-  diveRecoverTimer(),
-  state(FLYING),
-  last_player(nullptr),
-  last_player_pos(0.0f, 0.0f),
-  last_self_pos(0.0f, 0.0f)
+  m_catch_pos(0.f),
+  m_timer(),
+  m_state(FLYING)
 {
   m_physic.enable_gravity(false);
+  m_physic.set_velocity_x(FLYING_SPEED);
 }
 
 void
 Zeekling::initialize()
 {
-  m_physic.set_velocity_x(m_dir == Direction::LEFT ? -speed : speed);
+  m_physic.set_velocity_x(m_physic.get_velocity_x() * (m_dir == Direction::LEFT ? -1 : 1));
```

Comment:
I'm confused: When m_dir is not left, this becomes a no op. Can this not be moved to the constructor?

Reply:
No. When using the auto direction, auto gets turned into left or right before the initialize function is called. See https://github.com/MatusGuy/supertux/blob/zeekling_behavior/src/badguy/badguy.cpp#L828